src/script/script.cpp

```diff
@@ -255,6 +256,60 @@ bool CScript::IsPayToScriptHash() const
             (*this)[22] == OP_EQUAL);
 }
 
+bool CScript::IsRedeemScriptReveal(CScript scriptpubkey) const{
+    CScript check_spk, redeemScript = scriptpubkey;
+    if (
+        // these magic numbers correspond to a typical(as of May 2021) atomicdex HLTC
+        redeemScript[0] == OP_RETURN &&
+        redeemScript[3] == OP_IF &&
+        redeemScript[9] == OP_NOP2 &&
+        redeemScript[10] == OP_DROP &&
+        redeemScript[45] == OP_CHECKSIG &&
+        redeemScript[46] == OP_ELSE &&
+        redeemScript[47] == OP_SIZE &&
+        redeemScript[48] == 0x01 &&
+        redeemScript[49] == 0x20 &&
+        redeemScript[50] == OP_EQUALVERIFY &&
+        redeemScript[51] == OP_HASH160 &&
```

Comment:
I have also just noticed that there are coins that don't support RIPEMD160 natively in their smart contracts - e.g. Tezos (which wasn't completely integrated but we would like to revive it in the near future). To cover this case could you please also add an exception for `OP_SHA1`, `OP_SHA256`, and `OP_HASH256`? It also implies script length change as `OP_SHA256` and `OP_HASH256` will be followed by 32 bytes, not 20.

Reply:
Please provide me the exact script needed to be added as an exception

Reply:
OP_SHA256
```
OP_IF
OP_PUSHBYTES_4 0xd9b49b60
OP_CHECKLOCKTIMEVERIFY
OP_DROP
OP_PUSHBYTES_33 0x030101010101010101010101010101010101010101010101010101010101010101
OP_CHECKSIG
OP_ELSE
OP_SIZE
OP_PUSHBYTES_1 0x20
OP_EQUALVERIFY
OP_SHA256
OP_PUSHBYTES_32 0x0000000000000000000000000000000000000000000000000000000000000000
OP_EQUALVERIFY
OP_PUSHBYTES_33 0x030101010101010101010101010101010101010101010101010101010101010101
OP_CHECKSIG
OP_ENDIF
```

OP_HASH256
```
OP_IF
OP_PUSHBYTES_4 0xb6b59b60
OP_CHECKLOCKTIMEVERIFY
OP_DROP
OP_PUSHBYTES_33 0x030101010101010101010101010101010101010101010101010101010101010101
OP_CHECKSIG
OP_ELSE
OP_SIZE
OP_PUSHBYTES_1 0x20
OP_EQUALVERIFY
OP_HASH256
OP_PUSHBYTES_32 0x0000000000000000000000000000000000000000000000000000000000000000
OP_EQUALVERIFY
OP_PUSHBYTES_33 0x030101010101010101010101010101010101010101010101010101010101010101
OP_CHECKSIG
OP_ENDIF
```

As SHA1 is considered insecure I abandon the idea of adding `OP_SHA1`.

Reply:
Does OP_SHA256 alone cover all cases or is there a specific need for OP_HASH256?

Reply:
`OP_SHA256` should be sufficient, but I would prefer having `OP_HASH256` just in case, if possible.

Reply:
I prefer we have some tangible plans for any consensus code we activate. Adding it now means we cannot ever remove it.